src/framework/global/internal/interactive.cpp

```diff
@@ -224,9 +245,37 @@ async::Promise<IInteractive::Result> Interactive::error(const std::string& conte
     return openStandartAsync("ERROR", contentTitle, text, buttons, defBtn, options, dialogTitle);
 }
 
-Ret Interactive::showProgress(const std::string& title, Progress* progress) const
+void Interactive::showProgress(const std::string& title, Progress* progress)
+{
+    UriQuery q("muse://interactive/progress");
+    q.set("title", title)
+    .set("progress", Val((void*)progress));
```

Comment:
This is of course quite unsafe... I hope it is not _too_ unsafe (both for security and for bugs).
On the one hand, it's not very different from how it used to be; on the other hand, it is, because now we allow passing pointers via Uri strings. I'm not sure if we can be certain that this does not open vulnerabilities.

Reply:
I like option 2. Perhaps that also reduces the number of "back and forth" conversions between Qt types and Val etc.